Tycho: Cannot deal with automatic bundle wrapping

## Description

There is a public repository at https://github.com/wkl3nk/com.example.parent.git that can be used to reproduce the issue.

In `target-platform/target-platform.target` it has a section for dependencies of `type="Maven"` and `missingManifest="generate"`: https://github.com/wkl3nk/com.example.parent/blob/9ab48f8186f64c423ca0d54428f306a05b0daf1d/target-platform/target-platform.target#L17-L26

```xml
<location type="Maven" includeDependencyDepth="infinite" 
    includeDependencyScopes="compile" includeSource="true" 
    label="maven.libraries.with.transitives" missingManifest="generate">
    <dependencies>
        <dependency>
            <groupId>org.apache.xmlbeans</groupId>
            <artifactId>xmlbeans</artifactId>
            <version>3.1.0</version>
            <type>jar</type>
        </dependency>
    </dependencies>
</location>
```

Trying to analyze this project with ORT (72.1.1) results in the following messages:

```
% ort analyze -i . -o . 
Hoplite is configured to infer which sealed type to choose by inspecting the config values at runtime. This behaviour is now deprecated in favour of explicitly specifying the type through a discriminator field. In 3.0 this new behavior will become the default. To enable this behavior now (and disable this warning), invoke withExplicitSealedTypes() on the ConfigLoaderBuilder.
 ______________________________                                                        
/        \_______   \__    ___/ The OSS Review Toolkit, version 72.1.1-020.sha.ccbccc5,
|    |   | |       _/ |    |    built with JDK 21.0.8+9-LTS, running under Java 21.0.8.
|    |   | |    |   \ |    |    Executing 'analyze' as 'klw1imb' on Mac OS X           
\________/ |____|___/ |____|    with 14 CPUs and a maximum of 12288 MiB of memory.     
                                                                                       
Environment variables:                                                                
HOME = /Users/klw1imb                                                                 
SHELL = /bin/zsh                                                                      
TERM = xterm-256color                                                                 
JAVA_HOME = /Users/klw1imb/.sdkman/candidates/java/current                            
                                                                                      
Looking for ORT configuration in the following file:
        /Users/klw1imb/.ort/config/config.yml (does not exist)

Looking for analyzer-specific configuration in the following files and directories:
        /Users/klw1imb/workspace/com.example.parent/.ort.yml (does not exist)
        /Users/klw1imb/.ort/config/resolutions.yml (does not exist)
The following 26 package manager(s) are enabled:
        Bazel, Bower, Bundler, Cargo, Carthage, CocoaPods, Composer, Conan, GoMod, Gradle Inspector, Maven, NPM, NuGet, PIP, Pipenv, PNPM, Poetry, Pub, SBT, SpdxDocumentFile, Stack, Swift Package Manager, Tycho, Unmanaged, Yarn, Yarn 2+
The following 3 package curation provider(s) are enabled:
        DefaultDir, DefaultFile, Spring
Analyzing project path:
        /Users/klw1imb/workspace/com.example.parent
Found 1 Tycho definition file(s) at:
        pom.xml
Found in total 1 definition file(s) from the following 1 package manager(s):
        Tycho
18:11:26.857 [DefaultDispatcher-worker-1] WARN  org.apache.maven.shared.filtering.DefaultMavenResourcesFiltering - Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
18:11:26.865 [DefaultDispatcher-worker-1] WARN  org.eclipse.tycho.compiler.OsgiCompilerMojo - Source directory /Users/klw1imb/workspace/com.example.parent/com.example.hello/src/main/java does not exist
18:11:26.867 [DefaultDispatcher-worker-1] WARN  org.apache.maven.shared.filtering.DefaultMavenResourcesFiltering - Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
18:11:27.270 [DefaultDispatcher-worker-1] WARN  org.apache.maven.shared.filtering.DefaultMavenResourcesFiltering - Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
18:11:27.328 [DefaultDispatcher-worker-1] WARN  org.apache.maven.shared.filtering.DefaultMavenResourcesFiltering - Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
18:11:30.853 [DefaultDispatcher-worker-1] ERROR org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.MavenSupport - Failed to build 'p2.eclipse.plugin:org.eclipse.osgi:3.23.200.v20250812-1847': ProjectBuildingException: Error resolving project artifact: The following artifacts could not be resolved: p2.eclipse.plugin:org.eclipse.osgi:pom:3.23.200.v20250812-1847 (absent): Could not find artifact p2.eclipse.plugin:org.eclipse.osgi:pom:3.23.200.v20250812-1847 in central (https://repo.maven.apache.org/maven2) for project p2.eclipse.plugin:org.eclipse.osgi:pom:3.23.200.v20250812-1847
Caused by: ArtifactResolutionException: The following artifacts could not be resolved: p2.eclipse.plugin:org.eclipse.osgi:pom:3.23.200.v20250812-1847 (absent): Could not find artifact p2.eclipse.plugin:org.eclipse.osgi:pom:3.23.200.v20250812-1847 in central (https://repo.maven.apache.org/maven2)
    Caused by: ArtifactNotFoundException: Could not find artifact p2.eclipse.plugin:org.eclipse.osgi:pom:3.23.200.v20250812-1847 in central (https://repo.maven.apache.org/maven2)
18:11:31.386 [DefaultDispatcher-worker-1] ERROR org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.MavenSupport - Failed to build 'p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:3.1.0': ProjectBuildingException: Error resolving project artifact: The following artifacts could not be resolved: p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 (absent): Could not find artifact p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 in central (https://repo.maven.apache.org/maven2) for project p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0
Caused by: ArtifactResolutionException: The following artifacts could not be resolved: p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 (absent): Could not find artifact p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 in central (https://repo.maven.apache.org/maven2)
    Caused by: ArtifactNotFoundException: Could not find artifact p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 in central (https://repo.maven.apache.org/maven2)
18:11:31.387 [DefaultDispatcher-worker-1] ERROR org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.MavenDependencyHandler - Could not get package information for dependency 'p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:3.1.0': ProjectBuildingException: Error resolving project artifact: The following artifacts could not be resolved: p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 (absent): Could not find artifact p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 in central (https://repo.maven.apache.org/maven2) for project p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0
Caused by: ArtifactResolutionException: The following artifacts could not be resolved: p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 (absent): Could not find artifact p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 in central (https://repo.maven.apache.org/maven2)
    Caused by: ArtifactNotFoundException: Could not find artifact p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0 in central (https://repo.maven.apache.org/maven2)
Wrote analyzer result to '/Users/klw1imb/workspace/com.example.parent/analyzer-result.yml' (0.02 MiB) in 177.294750ms.
The analysis took 7.965648s.
Found 5 project(s) and 9 package(s) in total (not counting excluded ones).
Applied 0 curation(s) from 0 of 3 provider(s).
Resolved issues: 0 errors, 0 warnings, 0 hints.
Unresolved issues: 1 error, 0 warnings, 0 hints.
There is 1 unresolved issue with a severity equal to or greater than the WARNING threshold.
```

## First analysis

The `<location type="Maven">` element in the target definition file is used to add artifacts directly from a Maven repository (like Maven Central) to the project's target platform. 
 
This is a crucial feature for bridging the gap between the standard Java ecosystem (managed by Maven) and the OSGi/Eclipse ecosystem (managed by Tycho/PDE).

The `missingManifest="generate"` tells Tycho what to do if a downloaded artifact from Maven is a plain JAR file and not a proper OSGi bundle (i.e., it's missing the `META-INF/MANIFEST.MF` file). generate instructs Tycho to automatically create a valid OSGi manifest for it on the fly.

The package name `p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans` is a direct result of the `missingManifest="generate"` attribute.

When `missingManifest="generate"` is used, Tycho and the Eclipse PDE will dynamically create this OSGi manifest for any artifact that lacks one. This process is known as "wrapping" the standard JAR file to make it a usable OSGi bundle.

The Naming Convention: To signify that an artifact has been automatically converted, a prefix is added to its name.  By default, this prefix is `wrapped.`.

As this dependency is auto-generated, the ORT Maven package manager has no chance to find it on any repository and fails to find this dependency:
`p2.eclipse.plugin:wrapped.org.apache.xmlbeans.xmlbeans:pom:3.1.0`

--- a/plugins/package-managers/maven/src/main/kotlin/tycho/TargetHandler.kt
+++ b/plugins/package-managers/maven/src/main/kotlin/tycho/TargetHandler.kt
@@ -117,13 +117,18 @@ internal class TargetHandler(
      * to find the original Maven coordinates for affected artifacts. They are needed to retrieve the correct metadata.
      * Result is *null* if no matching Maven dependency is found.
      */
-    fun mapToMavenDependency(tychoArtifact: Artifact): Artifact? =
-        mavenDependencies[tychoArtifact.artifactId]?.also { dep ->
+    fun mapToMavenDependency(tychoArtifact: Artifact): Artifact? {
+        // Strip the "wrapped." prefix that might have been added by Tycho's automatic bundle wrapping mechanism
+        // (missingManifest="generate") to find the original Maven artifact.
+        val unwrappedArtifactId = tychoArtifact.artifactId.removePrefix("wrapped.")
+
+        return mavenDependencies[unwrappedArtifactId]?.also { dep ->
             logger.info {
                 "Mapping Tycho artifact '${tychoArtifact.groupId}:${tychoArtifact.artifactId}' to Maven " +
                     "dependency '${dep.groupId}:${dep.artifactId}'."
             }
         }
+    }
 }
 
 /**

--- a/plugins/package-managers/maven/src/test/assets/tycho.target
+++ b/plugins/package-managers/maven/src/test/assets/tycho.target
@@ -43,6 +43,15 @@
                     <version>2.15.2</version>
                     <type>jar</type>
                 </dependency>
+
+                <!-- This dependency has a missing manifest in its jar, so Tycho will generate one and wrap a
+                     p2 bundle around it. -->
+                <dependency>
+                    <groupId>org.apache.xmlbeans</groupId>
+                    <artifactId>xmlbeans</artifactId>
+                    <version>3.1.0</version>
+                    <type>jar</type>
+                </dependency>
             </dependencies>
         </location>
     </locations>

